### PR TITLE
[PyPer][BE] Fix test_scripted_module in StatCollector

### PR DESCRIPTION
**Pull request by PaulZhang12**
Summary: D41985889 removed the cast to int for the inputs to torch.histc below, allowing the inputs to still be tensors. These tensors still have require_grad_ set to True, causing issues with the call to torch.histc.

Test Plan: buck2 test 'fbcode//mode/opt' fbcode//dper3/dper3/modules/low_level_modules/tests:stat_collector_test -- --exact 'dper3/dper3/modules/low_level_modules/tests:stat_collector_test - test_scripted_module (dper3.dper3.modules.low_level_modules.tests.stat_collector_test.StatCollectorTest_1)'

Differential Revision: D48800879

### CODE REVIEW

**Review by jerryzh168 — APPROVED** [review]

File: torch/ao/quantization/observer.py [hunk 1]
```diff
@@ -1175,6 +1175,10 @@ def forward(self, x_orig: torch.Tensor) -> torch.Tensor:
             assert (
                 combined_min.numel() == 1 and combined_max.numel() == 1
             ), "histogram min/max values must be scalar."
+
+            # set requires_grad to False for these tensors for use in torch.histc
+            combined_min, combined_max = combined_min.detach(), combined_max.detach()
```
[the remark this answers is not in the document]
Comment (PaulZhang12) [reply torch/ao/quantization/observer.py:1180]: This is not related to x or x_orig. This is for combined_min and combined_max tensors to function below L1182

[the remark this answers is not in the document]
Comment (jerryzh168) [reply torch/ao/quantization/observer.py:1180]: oh, maybe we should detach `self.min_val` and `self.max_val` in L1146-1147?

[the remark this answers is not in the document]
Comment (PaulZhang12) [reply torch/ao/quantization/observer.py:1180]: Yeah, we could do that. We only run into this problem in this branch though, so figured just keep it here

[the remark this answers is not in the document]
Comment (jerryzh168) [reply torch/ao/quantization/observer.py:1180]: L1199 actually did this detach, so it might make sense to move this earlier I think

[the remark this answers is not in the document]
Comment (PaulZhang12) [reply torch/ao/quantization/observer.py]: detaching min_val/max_val did not work, as new tensors get introduced

**Review by jerryzh168 — APPROVED** [review]
